### ISSUE TITLE
Multiple bindings to a single source must have equivalent target values?

### ISSUE THREAD

**Issue by jda808** [jefffhaynes/BinarySerializer#17]
I'm baffled.

I'll update test case shortly.

````
System.InvalidOperationException was unhandled
  HResult=-2146233079
  Message=Error serializing type 'Pandacorn.Core.XXXX.ChunkContainer'.  See inner exception for detail.
  Source=BinarySerializer
  StackTrace:
       at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
       at BinarySerialization.BinarySerializer.Serialize(Stream stream, Object value, Object context) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\BinarySerializer.cs:line 130
       at xxx.xxxx.Program.Main(String[] args) in C:\s\xxx\xxxx\xxxx\Program.cs:line 163
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
       HResult=-2146233079
       Message=Error serializing type 'xxx.Core.XXXX.ChunkContainer'.  See inner exception for detail.
       Source=BinarySerializer
       StackTrace:
            at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
            at BinarySerialization.Graph.ValueGraph.ContextValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ContextValueNode.cs:line 85
            at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
       InnerException: 
            HResult=-2146233079
            Message=Error serializing member 'ChunkLength'.  See inner exception for detail.
            Source=BinarySerializer
            StackTrace:
                 at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                 at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
                 at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
            InnerException: 
                 HResult=-2146233079
                 Message=Error serializing member 'Chunk'.  See inner exception for detail.
                 Source=BinarySerializer
                 StackTrace:
                      at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                      at BinarySerialization.Graph.ValueGraph.ValueNode.MeasureOverride() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 216
                      at BinarySerialization.Graph.ValueGraph.ValueNode.<Bind>b__19_0() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 59
                      at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 45
                      at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
                      at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                 InnerException: 
                      HResult=-2146233079
                      Message=Error serializing member 'Chunks'.  See inner exception for detail.
                      Source=BinarySerializer
                      StackTrace:
                           at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                           at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
                           at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                      InnerException: 
                           HResult=-2146233079
                           Message=Error serializing type 'xxx.Core.XXXX.ChunkContainer'.  See inner exception for detail.
                           Source=BinarySerializer
                           StackTrace:
                                at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                                at BinarySerialization.Graph.ValueGraph.CollectionValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\CollectionValueNode.cs:line 27
                                at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                           InnerException: 
                                HResult=-2146233079
                                Message=Error serializing member 'ChunkLength'.  See inner exception for detail.
                                Source=BinarySerializer
                                StackTrace:
                                     at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                                     at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
                                     at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                                InnerException: 
                                     HResult=-2146233079
                                     Message=Error serializing member 'Chunk'.  See inner exception for detail.
                                     Source=BinarySerializer
                                     StackTrace:
                                          at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                                          at BinarySerialization.Graph.ValueGraph.ValueNode.MeasureOverride() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 216
                                          at BinarySerialization.Graph.ValueGraph.ValueNode.<Bind>b__19_0() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 59
                                          at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 45
                                          at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
                                          at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                                     InnerException: 
                                          HResult=-2146233079
                                          Message=Error serializing member 'ZoneCount'.  See inner exception for detail.
                                          Source=BinarySerializer
                                          StackTrace:
                                               at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
                                               at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
                                               at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                                          InnerException: 
                                               HResult=-2146233088
                                               Message=Multiple bindings to a single source must have equivalent target values.
                                               Source=BinarySerializer
                                               StackTrace:
                                                    at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 52
                                                    at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
                                                    at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
                                               InnerException:

**Comment by jefffhaynes** [comment]
Based on the error it looks like you may have multiple Subtype bindings
that point to different sources, which isn't currently allowed.

Message=Multiple bindings to a single source must have equivalent target values.


On Tue, May 5, 2015 at 3:13 PM, jda808 <notifications@github.com> wrote:

> I'm baffled.
>
> I'll update test case shortly.
>
> System.InvalidOperationException was unhandled
>   HResult=-2146233079
>   Message=Error serializing type 'Pandacorn.Core.XXXX.ChunkContainer'.  See inner exception for detail.
>   Source=BinarySerializer
>   StackTrace:
>        at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>        at BinarySerialization.BinarySerializer.Serialize(Stream stream, Object value, Object context) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\BinarySerializer.cs:line 130
>        at xxx.xxxx.Program.Main(String[] args) in C:\s\xxx\xxxx\xxxx\Program.cs:line 163
>        at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
>        at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
>        at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
>        at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
>        at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>        at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>        at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
>        at System.Threading.ThreadHelper.ThreadStart()
>   InnerException:
>        HResult=-2146233079
>        Message=Error serializing type 'xxx.Core.XXXX.ChunkContainer'.  See inner exception for detail.
>        Source=BinarySerializer
>        StackTrace:
>             at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>             at BinarySerialization.Graph.ValueGraph.ContextValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ContextValueNode.cs:line 85
>             at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>        InnerException:
>             HResult=-2146233079
>             Message=Error serializing member 'ChunkLength'.  See inner exception for detail.
>             Source=BinarySerializer
>             StackTrace:
>                  at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                  at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
>                  at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>             InnerException:
>                  HResult=-2146233079
>                  Message=Error serializing member 'Chunk'.  See inner exception for detail.
>                  Source=BinarySerializer
>                  StackTrace:
>                       at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                       at BinarySerialization.Graph.ValueGraph.ValueNode.MeasureOverride() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 216
>                       at BinarySerialization.Graph.ValueGraph.ValueNode.<Bind>b__19_0() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 59
>                       at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 45
>                       at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
>                       at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                  InnerException:
>                       HResult=-2146233079
>                       Message=Error serializing member 'Chunks'.  See inner exception for detail.
>                       Source=BinarySerializer
>                       StackTrace:
>                            at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                            at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
>                            at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                       InnerException:
>                            HResult=-2146233079
>                            Message=Error serializing type 'xxx.Core.XXXX.ChunkContainer'.  See inner exception for detail.
>                            Source=BinarySerializer
>                            StackTrace:
>                                 at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                                 at BinarySerialization.Graph.ValueGraph.CollectionValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\CollectionValueNode.cs:line 27
>                                 at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                            InnerException:
>                                 HResult=-2146233079
>                                 Message=Error serializing member 'ChunkLength'.  See inner exception for detail.
>                                 Source=BinarySerializer
>                                 StackTrace:
>                                      at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                                      at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
>                                      at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                                 InnerException:
>                                      HResult=-2146233079
>                                      Message=Error serializing member 'Chunk'.  See inner exception for detail.
>                                      Source=BinarySerializer
>                                      StackTrace:
>                                           at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                                           at BinarySerialization.Graph.ValueGraph.ValueNode.MeasureOverride() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 216
>                                           at BinarySerialization.Graph.ValueGraph.ValueNode.<Bind>b__19_0() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 59
>                                           at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 45
>                                           at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
>                                           at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                                      InnerException:
>                                           HResult=-2146233079
>                                           Message=Error serializing member 'ZoneCount'.  See inner exception for detail.
>                                           Source=BinarySerializer
>                                           StackTrace:
>                                                at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 132
>                                                at BinarySerialization.Graph.ValueGraph.ObjectValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ObjectValueNode.cs:line 112
>                                                at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                                           InnerException:
>                                                HResult=-2146233088
>                                                Message=Multiple bindings to a single source must have equivalent target values.
>                                                Source=BinarySerializer
>                                                StackTrace:
>                                                     at BinarySerialization.Graph.ValueGraph.ValueValueNode.get_BoundValue() in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 52
>                                                     at BinarySerialization.Graph.ValueGraph.ValueValueNode.SerializeOverride(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueValueNode.cs:line 64
>                                                     at BinarySerialization.Graph.ValueGraph.ValueNode.Serialize(StreamLimiter stream, EventShuttle eventShuttle) in C:\s\3rd\social\github\BinarySerializer\BinarySerializer\Graph\ValueGraph\ValueNode.cs:line 119
>                                                InnerException:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jefffhaynes/BinarySerializer/issues/17>.
>



-- 
If you want to build a ship, don't drum up people together to collect wood
and don't assign them tasks and work, but rather teach them to long for the
endless immensity of the sea.

Antoine de Saint-Exupery

**Comment by jefffhaynes** [comment]
Disregard that, I was thinking of the wrong thing.  This is actually indicative that you have two unrelated bindings pointing back to a single source field.  The serializer will attempt to update that source value during serialization and if it is updated by more than one binding, those bindings must be in agreement.  For example, this will not work if the length of Value1 does not equal the length of Value2 during serialization:

```C#
public class MyClass
{
    [FieldOrder(0)]
    public int Length { get; set; }

    [FieldOrder(1)]
    [FieldLength("Length")]
    public string Value1 { get; set; }

    [FieldOrder(2)]
    [FieldLength("Length")]
    public string Value2 { get; set; }
}

**Comment by jda808** [comment]
Aha! I was confused there for a minute.
Correct. In the Body chunk, there are two lists bound to "ZoneCount". My problem is/was occurring because I am 'injecting' values via another graph. I am failing to properly update the second list so they are out of whack. 

``` C#
	[FieldOrder(1)]
	public WankerVersion GroupVersion { get; set; } 

		[SerializeAs(Endianness = Endianness.Big)]
		[FieldOrder(2)]
		public UInt32 GroupCount { get; set; } = 0;


		[FieldOrder(3)]
		[FieldCount("GroupCount")]
		public List<GroupParameterData> Groups { get; set; } = new List<GroupParameterData>();

		/// <summary>
		/// Should be 2,2,0
		/// </summary>
		[FieldOrder(4)]
		[FieldLength(5)]
		public WankerVersion ZoneVersion { get; set; }
		[SerializeAs(Endianness = Endianness.Big)]
		[FieldOrder(5)]
		public UInt32 ZoneCount { get; set; } = 0;

		[FieldOrder(6)]
		[FieldCount("ZoneCount")]
		public List<ZoneParameterData> Zones { get; set; } = new List<ZoneParameterData>();

		[FieldOrder(7)]
		[FieldCount("ZoneCount")]
		public List<SampleRefData> SampleReferences { get; set; } = new List<SampleRefData>();

```

and the offending code is too pathetic to post, but essentially it doesn't update SampleReferences properly.

Once again, thank you for the clarification!

**Comment by jefffhaynes** [comment]
Cool, no worries

On Tue, May 5, 2015 at 7:03 PM, jda808 <notifications@github.com> wrote:

> Closed #17 <https://github.com/jefffhaynes/BinarySerializer/issues/17>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jefffhaynes/BinarySerializer/issues/17#event-297734466>
> .
>



-- 
If you want to build a ship, don't drum up people together to collect wood
and don't assign them tasks and work, but rather teach them to long for the
endless immensity of the sea.

Antoine de Saint-Exupery
